// botbuilder-botbldr/lib/ServiceBot.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const botbuilder_1 = require("botbuilder");
const botbldr_1 = require("./botbldr");
const restify_1 = require("restify");
class ServiceBot extends botbldr_1.StateBot {
    constructor(storage) {
        super(storage);
        this.adapter = new botbuilder_1.BotFrameworkAdapter({
            appId: process.env.MICROSOFT_APP_ID,
            appPassword: process.env.MICROSOFT_APP_PASSWORD
        })
            .use(this.conversationState)
            .use(this.userState);
    }
    onRequest(handler) {
        this.adapter.use(...this.middlewares);
        const server = restify_1.createServer();
        server.post('/api/messages', (req, res) => {
            this.adapter.processRequest(req, res, this.do(handler));
        });
        server.listen(process.env.port || process.env.PORT || 3978, () => {
            console.log(`${server.name} listening to ${server.url}`);
        });
    }
    startConversation(reference, handler) {
        return this.adapter.startConversation(reference, this.do(handler));
    }
    // when startConversation moves into Adapter, this can be implemented here
    continueConversation(reference, handler) {
        return this.adapter.continueConversation(reference, this.do(handler));
    }
}
exports.ServiceBot = ServiceBot;
class StarterServiceBot extends ServiceBot {
    constructor() {
        super(new botbuilder_1.MemoryStorage());
    }
}
exports.StarterServiceBot = StarterServiceBot;
